ui/src/app/base/components/ControllerLink/ControllerLink.tsx

```diff
@@ -1,16 +1,38 @@
+import { Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
 import LegacyLink from "app/base/components/LegacyLink";
 import baseURLs from "app/base/urls";
-import type { BaseController } from "app/store/controller/types/base";
+import controllerSelectors from "app/store/controller/selectors";
+import type { Controller, ControllerMeta } from "app/store/controller/types";
+import type { RootState } from "app/store/root/types";
+
+type Props = {
+  systemId?: Controller[ControllerMeta.PK] | null;
+};
+
+const ControllerLink = ({ systemId }: Props): JSX.Element | null => {
```

Comment:
Should this also fetch the controllers to make sure they exist?

Reply:
Yeah, since this exists in base components it should probably just handle everything